plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -566,9 +568,23 @@ def initialize(self):
         self.context = UpdateinfoXMLFileContext(self.get_working_dir(), checksum_type,
                                                 self.get_conduit())
         self.context.initialize()
+
+        repo_id = self.get_repo().id
+        rpm_content_type = models.RPM._content_type_id.default
+        nevra_fields = ('name', 'epoch', 'version', 'release', 'arch')
+        unit_ids_list = list(repo_controller.get_associated_unit_ids(repo_id, rpm_content_type))
+        repo_nevra_q = models.RPM.objects.filter(id__in=unit_ids_list).scalar(*nevra_fields)
+        repo_nevra = []
+        for scaler in repo_nevra_q:
+            repo_nevra.append(models.NEVRA(*scaler))
+
+        def curry_add_unit_metadata(item):
+            """Wrap the actual method. Causes repo_nevra to be passed along on all future calls."""
+            self.context.add_unit_metadata(item, repo_nevra)
+
         # set the self.process_unit method to the corresponding method on the
         # UpdateInfoXMLFileContext as there is no other processing to be done for each unit.
-        self.process_main = self.context.add_unit_metadata
+        self.process_main = curry_add_unit_metadata
```

Comment:
Since this method creates the UpdateInfoxmlFileContext instance, about 20 lines above, what if you just pass the ``repo_nevra`` object into that class's ``__init__``, and it can save it as an instance attribute. I think that would be easier to follow for future readers of this code.